capstone/src/main/webapp/youtubeScript.js

```diff
@@ -270,3 +271,32 @@ function showCategories() {
     container.appendChild(document.createTextNode (" "));
   }
 }
+
+/** Converts perspective results to knoop scale than to mohs*/
+function perspectiveToxicityScale(attributeAverages) {
+  const knoopScale = [1, 32, 135, 163, 430, 560, 820, 1340, 1800, 7000];
+  var totalToxicityScore = 0;
+  for (const [attribute, attributeAverage] of attributeAverages) {
+    totalToxicityScore += attributeAverage;
+  }
+  const inputLength = attributeAverages.size;
+  const averageToxicityScore = totalToxicityScore / inputLength;
+  const knoopScore = averageToxicityScore * 7000;
+  var knoopLow;
+  var knoopHigh;
+  var mohs;
+  for (var i = 0; i < knoopScale.length; i++) {
+    if (knoopScore < knoopScale[i]) {
+      knoopLow = knoopScale[i-1];
+      knoopHigh = knoopScale[i];
+      mohsScore = i;  
+      break;
+    }
+  }
```

Comment:
Lines 284-295 could run into an error if `knoopScore` is less than 32 because of line 290 (when `i = 0`, there is no `knoopScale[-1]`).

Depending on if we want to ratings to run from 0 to 10 or 1 to 10, we may also want to make `knoopscale = [0, 1, 32...` because `knoopScore` currently can be anywhere from 0 to 7000.

Reply:
addressed in "address edge case for toxicity scale"
We can certainly discuss why we may or may not want our scale to begin from 0.